uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/network/NetworkStateChangeHandler.java

```diff
@@ -0,0 +1,116 @@
+            mNetworkAvailable = false;
+        }
+    };
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    public NetworkStateChangeHandler() {
+        mNetworkAvailable = connectivityManager.getActiveNetworkInfo().isConnectedOrConnecting();
+        connectivityManager.registerDefaultNetworkCallback(networkCallback);
+    }
+
+    private boolean waitForNetworkAvailability(boolean networkAvailable) {
+        final String methodName = ":waitForNetworkAvailability";
+
+        Logger.i(TAG + methodName, "Waiting for network availability: " + networkAvailable);
+        if (networkAvailable != mNetworkAvailable) {
+            final CountDownLatch latch = new CountDownLatch(1);
+            try {
+                new Thread(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            while (networkAvailable != mNetworkAvailable) {
+                                Thread.sleep(1000);
+                            }
+                            latch.countDown();
+                        } catch (InterruptedException exception) {
+                            latch.countDown();
+                        }
+                    }
+                }).start();
+
+                latch.await(NETWORK_AVAILABILITY_WAIT_SECONDS, TimeUnit.SECONDS);
+            } catch (InterruptedException exception) {
+                Logger.e(TAG + methodName, "Error while waiting for network availability", exception);
+            }
+        }
+
+        Logger.i(TAG + methodName, "Network availability confirmed: " + (networkAvailable == mNetworkAvailable));
+        return networkAvailable == mNetworkAvailable;
+    }
+
+    private boolean changeNetworkState(NetworkInterface interfaceType) {
+        final String methodName = ":changeNetworkState";
+        if (interfaceType == null) return false;
+
+        Logger.i(TAG + methodName, "Changing network state to: " + interfaceType.getKey());
+
+        AdbShellUtils.executeShellCommand("svc wifi " + (interfaceType.wifiActive() ? "enable" : "disable"));
+        AdbShellUtils.executeShellCommand("svc data " + (interfaceType.cellularActive() ? "enable" : "disable"));
+
+        return waitForNetworkAvailability((interfaceType.cellularActive() || interfaceType.wifiActive()));
+
+    }
+
+    @Override
+    public boolean onNetworkStateApplied(@NonNull NetworkMarker marker, @NonNull NetworkState networkState) {
+        final NetworkInterface networkInterface = networkState.getNetworkInterface();
+
+        return changeNetworkState(networkInterface);
+    }
+
+    @Override
+    public boolean onRestoreNetworkState(@Nullable NetworkMarker marker, @Nullable NetworkState networkState) {
+        return changeNetworkState(NetworkInterface.WIFI_AND_CELLULAR);
+    }
+
+    @Override
+    public void handleNetworkStateNotApplied(@NonNull NetworkMarker marker, @NonNull NetworkState networkState) {
```

Comment:
Looks like there is no implementation of this method.  What do you want to have happen here?  Should we fail the test?